timemachine/cpp/src/device_buffer.hpp

```diff
@@ -9,9 +9,13 @@ template <typename T> class DeviceBuffer {
 
     ~DeviceBuffer();
 
-    const size_t size;
+    size_t length;
```

Comment:
maybe document the difference between length and size? I'm guessing `size=length*sizeof(T)`?

Reply:
(and isn't size always inferrable from length?)

Reply:
Moved to method `size()` https://github.com/proteneer/timemachine/pull/1196/commits/c1ca8f0d6f62b79437eec92d4bb031c5d2de69b9

Reply:
Note that size() in `std::vector` and other `std::container` classes returns number of elements, where as we're more consisten with the cudaMalloc definition of `size`, i.e. `size = number of elements * size of each element`, please be sure to document this inconsistency.

Reply:
Added docstring https://github.com/proteneer/timemachine/pull/1196/commits/2e1f7b82c8d0e723592091eb119431b552e3b96e